OpenRA.Mods.Common/Activities/UnloadCargo.cs

```diff
@@ -121,10 +121,10 @@ public override bool Tick(Actor self)
 					var move = actor.Trait<IMove>();
 					var pos = actor.Trait<IPositionable>();
 
-					pos.SetPosition(self, exitSubCell.Value.First, exitSubCell.Value.Second);
+					pos.SetPosition(actor, exitSubCell.Value.First, exitSubCell.Value.Second);
 					pos.SetVisualPosition(actor, spawn);
 
-					actor.CancelActivity();
+					// actor.CancelActivity();
```

Comment:
This seems to cancel the `MoveIntoWorld` activity, making the actor visually appear at a different location than it actually is, causing the glitch seen in #17009.

Reply:
Removing this has no drawbacks in my initial tests and it actually allows actors to perform tasks queued after a load activity.

Reply:
This won't cancel the `MoveIntoWorld` activity because `MoveIntoWorld` is not interruptible. It will cancel all activities queued afterwards, which was the idea.

Allowing units to perform tasks after hitching a ride was actually one of my motivations for implementing it in this way, because it would allow automated transports analogous to autocarryalls. However, this should be implemented by making this line dependent on a flag in the `UnloadCargo` constructor. By default it should cancel the activity queue (at least for now) because it would otherwise mess with the functionality to mass load into transports by shift queueing. (We should probably come up with a better mechanic for that at some point).

Reply:
Something is going wrong here though, restoring this line again results in the glitch:
![grafik](https://user-images.githubusercontent.com/7704140/65350906-13586800-dbe7-11e9-8f14-e3113ec2b95d.png)

Notice how the red overlay marks the spot the harvester is actually at.

Reply:
Ok, so apparently the harvesters activity at the point of cancelling is a interruptible `CallFunc` activity. Canceling this activity does null the `NextActivity` which is the uninterruptible `MoveIntoWorld` one.

Reply:
I am not sure how to resolve this. Keeping the queue may result in other unwanted behaviour.

Reply:
The best short-term fix for now is to just drop the `CallFunc` in `Harvester`. That one is completely unnecessary and thus should be removed anyway.